PyTado/interface/api/hops_tado.py

```diff
@@ -324,3 +324,17 @@ def set_temp_offset(self, device_id, offset=0, measure="celsius"):
         request.payload = {"temperatureOffset": offset}
 
         return self._http.request(request)
+
+    def set_child_lock(self, device_id, child_lock):
+        """ "
+        Set and toggle the child lock on the device.
+        """
+
+        request = TadoXRequest()
+        request.command = "childLock"
+        request.action = Action.CHANGE
+        request.device = device_id
+        request.domain = Domain.DEVICES
+        request.payload = {"childLockEnabled": child_lock}
```

Comment:
I guess this won't work :/

This would create the URL `https://hops.tado.com/devices/<device_id>/childLock` and not `https://hops.tado.com/homes/1745644/roomsAndDevices/devices/VA3250358016`. I think we have to make changes to `http.py` for this too.

Reply:
I might have applied to this a bit too fast and copied this from another HOPS:

```py
    def get_device_info(self, device_id, cmd=""):
        """
        Gets information about devices
        with option to get specific info i.e. cmd='temperatureOffset'
        """

        if cmd:
            request = TadoRequest()
            request.command = cmd
        else:
            request = TadoXRequest()

        request.action = Action.GET
        request.domain = Domain.DEVICES
        request.device = device_id

        return self._http.request(request)
```

We need bigger changes for this, indeed. I will work this out this evening.